### PR TITLE
PPU render_sprite_pixel: rudimentary optimization

### PR DESCRIPTION
**Pull request by Optimoos**
Did some very simplistic optimization of this function. I'm not happy about the unchecked math for sprite_attr but when profiling, these lines were quite hot. There is certainly a better way to optimize this function specifically, but there may also be optimizations possible in Jakt itself. I'm just here as a hobbyist programmer though so it's going to take me a lot more time to wrap my head around all of this.

These changes shaved ~5ms off my frame times, bringing them from ~20-25ms to ~15-20ms.

### CODE REVIEW

File: src/ppu.jakt [hunk 1]
```diff
@@ -139,10 +139,7 @@ class PPU {
 
     function render_sprite_pixel(mut this, horizontal_pixel: u64, vertical_pixel: u64, is_behind: bool) throws -> bool {
         for sprite in 63..-1 {
-            let sprite_y = .oam[sprite * 4] as! u64
-            let sprite_index = .oam[sprite * 4 + 1] as! u8
-            let sprite_attr = .oam[sprite * 4 + 2] as! u8
-            let sprite_x = .oam[sprite * 4 + 3] as! u64
+            let sprite_attr = .oam[unchecked_add(unchecked_mul(sprite,4),2)]
```
Comment (sophiajt) [inline src/ppu.jakt]: this is done for you automatically when you build in release mode, thankfully

Comment (Optimoos) [reply src/ppu.jakt]: Still struggle with git sometimes, but I think I managed to amend this to remove the explicit unchecked adds.

File: src/ppu.jakt [hunk 2]
```diff
@@ -179,10 +176,22 @@ class PPU {
...
+            let sprite_x = .oam[sprite * 4 + 3] as! u64
+
             // TODO: handle 8x16 sprites
             if (vertical_pixel >= sprite_y) and (vertical_pixel < (sprite_y + 8)) and (horizontal_pixel >= sprite_x) and (horizontal_pixel < (sprite_x + 8)) {
                 mut sprite_row = (vertical_pixel - sprite_y) as! u64
                 mut sprite_col = (horizontal_pixel - sprite_x) as! u64
+                                
+                let sprite_index = .oam[sprite * 4 + 1]
+
+                let sprite_index = .oam[sprite * 4 + 1]
```
Comment (sophiajt) [inline src/ppu.jakt]: and also here?

File: src/ppu.jakt [hunk 3]
```diff
@@ -179,10 +176,22 @@ class PPU {
                 continue
             }
 
+            let sprite_y = .oam[sprite * 4] as! u64
+
+            let sprite_x = .oam[sprite * 4 + 3] as! u64
+
+            let sprite_y = .oam[sprite * 4] as! u64
+
+            let sprite_x = .oam[sprite * 4 + 3] as! u64
```
Comment (sophiajt) [inline src/ppu.jakt]: it looks like you're duplicating the assignment here?

**Comment by Optimoos** [comment]
Should be better now

**Pull request merged**